modules/other/vm_with_power_automate_desktop/main.tf

```diff
@@ -0,0 +1,133 @@
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.48"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.5"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.1"
+    }
+    http = {
+      source  = "hashicorp/http"
+      version = "~> 2.1"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = "~> 2.1"
+    }
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_public_ip" "public_ip" {
+  count = var.enable_public_access ? 1 : 0
+
+  name                = "public-ip-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  allocation_method   = "Dynamic"
+}
```

Comment:
By using a Dynamic allocation_method the vm can have an other IP address assigned when it got rebooted. This can led into an issue when whitelisting is being used. Suggested change:

allocation_method = "Static"
sku = "Standard"

Reply:
Good one!